src/meta/processors/admin/Balancer.h

```diff
@@ -219,7 +228,10 @@ class Balancer {
     std::unique_ptr<HostLeaderMap> hostLeaderMap_;
     mutable std::mutex lock_;
 
+    int32_t totalParts_;
+    std::unordered_map<HostAddr, int32_t> partsDistribution_;
     std::unordered_map<HostAddr, std::pair<int32_t, int32_t>> hostBounds_;
+    std::unordered_map<HostAddr, ZoneNameAndParts> zoneParts_;
```

Comment:
Emm... `<HostAddr, <Zone name, list of part Id>`?

Reply:
yep